language/extensions/move-table-extension/sources/Table.move

```diff
@@ -1,17 +1,26 @@
 /// Type of large-scale storage tables.
 module Extensions::Table {
+    use Std::Errors;
+
+    const ENOT_EMPTY: u64 = 0;
+
     /// Type of tables
     struct Table<phantom K, phantom V> has store {
-        handle: u128
+        handle: u128,
+        length: u64,
     }
 
     /// Create a new Table.
     public fun new<K, V: store>(): Table<K, V> {
-        Table{handle: new_table_handle()}
+        Table{
+            handle: new_table_handle(),
+            length: 0,
+        }
     }
 
     /// Destroy a table. The table must be empty to succeed.
     public fun destroy_empty<K, V>(table: Table<K, V>) {
+        assert!(table.length == 0, Errors::invalid_state(ENOT_EMPTY));
```

Comment:
I think we still may want to return a unique abortion code here, as it is/was defined in the Rust code. You could have a native function here to discover this abortion code.